### PR TITLE
document modals + issue reporting

### PR DESCRIPTION
**Pull request by davidsmejia**
## Issue Number

#460 

## Purpose/Implementation Notes

Adds in a little more than intended.

`View All Request Documents` and associated modals. Also includes a refactor so the logic is more dry and the rendering of the associated part of the modal are more consistent when partially show outside of the modal.

Also adds the issue reporting functionality. Sorry about that.

## Types of changes

- Bugfix (non-breaking change which fixes an issue)
- New feature (non-breaking change which adds functionality)

## Functional tests

Tested on all steps of the request process from both sharer and requester points of view.

## Checklist

- [x] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

## Screenshots

Screen Shot 2020-10-05 at 12 54 32 AM

### CODE REVIEW

**Comment by davidsmejia** [comment]
The `created_at` attribute needs to be added to the serializer for fulfillment notes for that to work... (see screenshot)

File: client/src/helpers/getRequestAwaitingAdditionalDocuments.js [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+  const {
+    needsIrb,
+    needsMta,
+    shippingRequirement: { needsPayment }
+  } = getRequestRequirements(request.material)
+
+  const additionalDocuments = []
+  if (needsIrb) additionalDocuments.push('IRB')
+  if (needsMta) additionalDocuments.push('signed MTA')
+  if (needsPayment) additionalDocuments.push('shiping payment method')
```
Comment (kurtwheeler) [inline client/src/helpers/getRequestAwaitingAdditionalDocuments.js:13]: Is this only used for new requests? You're pulling these fields off the material, so the current state of the request doesn't impact them at all.

Comment (davidsmejia) [reply client/src/helpers/getRequestAwaitingAdditionalDocuments.js:13]: Screen Shot 2020-10-05 at 11 02 06 AM

File: client/src/components/resources/SharerManageRequestForm.js [hunk 2]
```diff
@@ -107,16 +127,28 @@ export default ({ request: defaultRequest }) => {
...
   }
 
+  const awaitingMtaSteps = []
+  if (state === 'AWAITING_MTA') {
+    if (needsIrb || needsPayment) {
+      awaitingMtaSteps.push(
+        'Review the additional documents submitted by the requests'
+      )
+    }
+    awaitingMtaSteps.push('Sign and upload the requester signed MTA')
```
Comment (kurtwheeler) [inline client/src/components/resources/SharerManageRequestForm.js:137]: This is the sharer view, should this say `Sign and upload the fully exectuted MTA`?

File: client/src/helpers/getRequestAwaitingAdditionalDocuments.js (hunk 1 above)
Comment (kurtwheeler) [reply client/src/helpers/getRequestAwaitingAdditionalDocuments.js:13]: That makes sense, although the word `awaiting` in the name seems a bit confusing given that the request hasn't even been made yet.

**Review by kurtwheeler — APPROVED** [review]
This LGTM! Please wait just a bit to merge cause I just merged #483

File: client/src/components/resources/SharerManageRequestForm.js (hunk 2 above)
Comment (davidsmejia) [reply client/src/components/resources/SharerManageRequestForm.js:137]: This is the copy from the comps but i do agree maybe @dvenprasad can weigh in?

File: client/src/helpers/getRequestAwaitingAdditionalDocuments.js (hunk 1 above)
Comment (davidsmejia) [reply client/src/helpers/getRequestAwaitingAdditionalDocuments.js]: yeah i agree i changed the name of the helper to `getRequesterWillBeAskedToProvide`

File: client/src/components/resources/SharerManageRequestForm.js (hunk 2 above)
Comment (dvenprasad) [reply client/src/components/resources/SharerManageRequestForm.js:137]: Yeah let's change it.

**Pull request merged**